# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/SharedFontList.h
# Commit: ac4ab326e785
# Full Hash: ac4ab326e785a5ad42ce51b54749c2bcce394916
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2024-02-19 21:22:25
# Description:
#   Bug 1879825 - Make Family::FindIndex fallible and safely handle failures. r=gfx-reviewers,lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D201819
# ==============================================================================

diff -r e4fcd942953c -r ac4ab326e785 gfx/thebes/SharedFontList.h
--- a/gfx/thebes/SharedFontList.h	Mon Feb 19 12:24:19 2024 +0000
+++ b/gfx/thebes/SharedFontList.h	Mon Feb 19 12:37:33 2024 +0000
@@ -333,7 +333,8 @@
 
   // Return the index of this family in the font-list's Families() or
   // AliasFamilies() list, and which of those it belongs to.
-  std::pair<uint32_t, bool> FindIndex(FontList* aList) const;
+  // Returns Nothing if the family cannot be found.
+  mozilla::Maybe<std::pair<uint32_t, bool>> FindIndex(FontList* aList) const;
 
  private:
   // Returns true if there are specifically-sized bitmap faces in the list,
